Classes/Event/EventPublisher.php

```diff
@@ -93,7 +94,8 @@ public function publishMany(string $streamName, array $events, int $expectedVers
             $metadata = [];
             $this->emitBeforePublishingEvent($event, $metadata);
             $data = $this->propertyMapper->convert($event, 'array');
-            $convertedEvents->append(new WritableEvent($type, $data, $metadata));
+            $eventIdentifier = Algorithms::generateUUID();
```

Comment:
At this point we could add sth like
```php
if ($event instanceof EventWithIdentifier) {
  $eventIdentifier = $event->getIdentifier();
}
```
but that clashes with the `AggregateEventInterface` and it could be added in a separate PR